src/index.js

```diff
@@ -1,41 +1,72 @@
 import './index.css';
-import addTask from './addtask.js';
-import { saveTasksToStorage, loadTasksFromStorage } from './localstorage.js';
+import {
+  addTask, removeTask, editTask, checkTasks, checkedBox, uncheckedBox, clearTasks,
+} from './modules/tasks.js';
 
-let tasks = loadTasksFromStorage();
+const lists = document.querySelector('.lists');
+const addBtn = document.querySelector('#add-btn');
+const resetBtn = document.querySelector('.reset-btn');
 
-const taskList = document.getElementById('taskList');
-const taskInput = document.getElementById('input');
+checkTasks();
 
-function addTasks() {
-  taskList.innerHTML = '';
-  tasks.forEach((task) => {
-    addTask(task, taskList, tasks);
-  });
-}
+addBtn.addEventListener('click', () => {
+  const description = document.getElementById('add-input').value;
+  addTask(description);
+  lists.insertAdjacentHTML(
+    'beforeend',
+    `<li class="task-item" contentEditable = "false"><input type="checkbox">${description}<i id="task-btn" class="fa-solid fa-ellipsis-vertical"></i></li>`,
+  );
+  document.getElementById('add-input').value = '';
+});
 
-taskInput.addEventListener('keyup', (event) => {
-  if (event.key === 'Enter') {
-    event.preventDefault();
-    const description = taskInput.value.trim();
+document.body.addEventListener('click', (e) => {
+  if (e.target.classList.contains('fa-trash')) {
+    const lis = document.querySelectorAll('.task-item');
+    let index = 0;
+    for (let i = 0; i < lis.length; i += 1) {
+      if (lis[i].textContent === e.target.parentElement.textContent) {
+        index = i;
+      }
+    }
+    e.target.parentElement.remove();
+    removeTask(index);
+  }
 
-    if (description !== '') {
-      const newTask = {
-        description,
-        completed: false,
-        index: tasks.length + 1,
-      };
+  if (e.target.classList.contains('fa-ellipsis-vertical')) {
+    e.target.parentElement.contentEditable = 'true';
+    e.target.classList.remove('fa-ellipsis-vertical');
+    e.target.classList.add('fa-trash');
 
-      tasks.push(newTask);
-      taskInput.value = '';
-      addTask(newTask, taskList, tasks);
+    document.querySelectorAll('.task-item').forEach((task) => {
+      const index = [...task.parentElement.children].indexOf(task) - 1;
 
-      saveTasksToStorage(tasks);
-    }
+      task.addEventListener('input', (e) => {
+        const description = e.target.textContent;
+        editTask(index, description);
+      });
+    });
   }
+  document.querySelectorAll('.task-item').forEach((task) => {
+    const index = [...task.parentElement.children].indexOf(task) - 1;
+    task.addEventListener('change', () => {
+      const myCheck = task.firstElementChild;
+
+      if (myCheck.checked) {
+        checkedBox(index);
+      } else {
+        uncheckedBox(index);
+      }
+    });
+  });
```

Comment:
👍 You've implemented the functionality to update the status of the todos' "completed" key values, which is a great step forward. However, there are some adjustments needed to improve the user experience:

- When the checkbox is clicked, the visual appearance of the related todo's task should update to reflect the change, as shown in the design. Please ensure that the design specifications are closely followed, and the visual appearance of the task updates appropriately upon checkbox interaction.

| Your | Design |
| --- | --- |
| ![image](https://github.com/newtonah0/todo-list/assets/55840999/94cf9e82-5c7a-4d80-be20-df14b3ea7c79) | ![image](https://github.com/newtonah0/todo-list/assets/55840999/0a26dca4-b894-452d-8b1c-95778a4b0a66) |

- Second, while you are storing the updated array of items in local storage, the completed and non-completed tasks' lists are not distinguishable visually upon reloading the page. To enhance user clarity, consider updating the visual appearance upon reloading the page to clearly differentiate completed and non-completed tasks, as indicated in the design.

[To-Do-List.webm](https://github.com/newtonah0/todo-list/assets/55840999/d7bb0337-5317-4ff6-9d42-82fa60512222)

- Additionally, After deleting the completed tasks once, if the user wants to mark the last element of the rest tasks, there is an error showing "Cannot set properties of undefined". This may be happening because your tasklist is not updating correctly. please conder to update your code to fix this error. 🙏 

[To-Do-List (1).webm](https://github.com/newtonah0/todo-list/assets/55840999/e8e0bd6d-dab3-41a5-9029-dd693ab3584e)

By addressing these adjustments, your project will provide a more intuitive and user-friendly experience. Keep up the good work, and I'm excited to see your project become even more polished! 🌟👁️‍🗨️🚀